api/envoy/config/core/v3/protocol.proto

```diff
@@ -345,6 +345,22 @@ message Http2ProtocolOptions {
   // <https://www.iana.org/assignments/http2-parameters/http2-parameters.xhtml#settings>`_ for
   // standardized identifiers.
   repeated SettingsParameter custom_settings_parameters = 13;
+
+  // Send HTTP/2 PING frames at this period, in order to test that the connection is still alive.
+  // If not specified, no keepalive PING frames will be sent.
+  google.protobuf.Duration connection_keepalive_interval = 15
+      [(validate.rules).duration = {gt {nanos: 1000000}}];
+
+  // How long to wait for a response to a keepalive PING. If a response is not received within this
+  // time period, the connection will be aborted.
+  //
+  // If configured, this value must be less than :ref:`connection_keepalive_interval
+  // <envoy_v3_api_field_config.core.v3.Http2ProtocolOptions.connection_keepalive_interval>`.
```

Comment:
Why must this be less? Can't it work similar to health checking where the interval and the timeout are disjoint? It seems like that might be what we want?

Reply:
I did this for ease of implementation (so there can't be two outstanding pings at the same time). I figured it would easy to relax this at some point in the future.

Reply:
I'm not sure how you can guarantee that no matter what you do, due to event loop delay, etc. It seems like it would be better if it worked like health checking where there is an outstanding ping with a timeout, and then an interval.

Reply:
Yeah, I agree that doing it like the health checker will be better (and more consistent). I'll go that way.